### DISCUSSION TITLE
ML container - Cache directory does not exist

### DISCUSSION THREAD

**Discussion by shackrock** [immich-app/immich#11250] [Q&A] [answered]
I'm running immich and ML containers via the docker on the website on QNAP.  I continue to get this error about the cache directory not existing (it does, with permissions read/write to public), and haven't yet found a fix, any ideas from anyone or how to troubleshoot next?  

After this error, it's just countless errors on every object/photo with status 500: internal server error over and over again.

Thanks.

```
immich_machine_learning  | [07/20/24 18:07:25] WARNING  Failed to load detection model 'buffalo_l'.        
immich_machine_learning  |                              Clearing cache.                                    
immich_machine_learning  | [07/20/24 18:07:25] WARNING  Attempted to clear cache for model 'buffalo_l', but
immich_machine_learning  |                              cache directory does not exist                     
immich_server            | [Nest] 7  - 07/20/2024, 6:07:25 PM   ERROR [Microservices:JobService] Unable to run job handler (faceDetection/face-detection): Error: Machine learning request '{"facial-recognition":{"detection":{"modelName":"buffalo_l","options":{"minScore":0.7}},"recognition":{"modelName":"buffalo_l"}}}' failed with status 500: Internal Server Error
immich_server            | [Nest] 7  - 07/20/2024, 6:07:25 PM   ERROR [Microservices:JobService] Error: Machine learning request '{"facial-recognition":{"detection":{"modelName":"buffalo_l","options":{"minScore":0.7}},"recognition":{"modelName":"buffalo_l"}}}' failed with status 500: Internal Server Error
immich_server            |     at MachineLearningRepository.predict (/usr/src/app/dist/repositories/machine-learning.repository.js:22:19)
immich_server            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
immich_server            |     at async MachineLearningRepository.detectFaces (/usr/src/app/dist/repositories/machine-learning.repository.js:33:26)
immich_server            |     at async PersonService.handleDetectFaces (/usr/src/app/dist/services/person.service.js:278:52)
immich_server            |     at async /usr/src/app/dist/services/job.service.js:148:36
immich_server            |     at async Worker.processJob (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:394:28)
immich_server            |     at async Worker.retryIfFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:581:24)
immich_server            | [Nest] 7  - 07/20/2024, 6:07:25 PM   ERROR [Microservices:JobService] Object:
immich_server            | {
immich_server            |   "id": "18b4ff1c-b3f6-4437-b485-988d1fabf213"
immich_server            | }
```

**Comment by PLCreed** [comment]
Hello, I am still getting the same error :(

**Reply by mertalev** [reply]
If you delete the model-cache volume, does the error persist?

**Reply by shackrock** [reply]
> If you delete the model-cache volume, does the error persist?

Delete the volume from the docker compose? or the actual folder on the NAS?  The folder on the NAS is literally empty for me. I have also tried giving it full public permissions hoping that would help, but nothing fixes it so far.  I assume it is something special/specific about QNAP, but I still haven't figured out what it could be. Perhaps the docker virtual network that is created automatically is not allowing traffic to talk between them? But this is beyond my troubleshooting capability if so.

**Comment by erwinsonp** [comment]
Are you perhaps running a Pi-hole in a container in your network?
I had an unrelated problem with Wireguard that it didn't like the Pi-hole running inside a container, and making a fix in Pi-hole port bindings also solved this problem in my Immich install.

**Reply by shackrock** [reply]
I don't have pi-hole, but there is a firewall on qnap, though I am honestly not sure where to begin or what to try there if it could be related.

**Comment by shackrock** [comment, marked as answer]
Well, I just updated from an older version of Immich to the latest 1.120.1, and for whatever reason it works without an issue now on QNAP. I did not change ANYTHING in the ML docker from my version upgrades, so I really don't know what the issue was, sorry.  But maybe everyone will be solved now by upgrading to the latest.
